### PR TITLE
Upgrade to python3 / fix build on 12.3

### PR DESCRIPTION
**Pull request by garrettmoon**
Fixes the build on 12.3 and updates python scripts to /usr/bin/python3

This doesn't upgrade all the example projects - some started to fail.

### CODE REVIEW

**Review by rgnns — APPROVED** [review]

File: BazelExtensions/headermap_tool.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ class HeaderMap(object):
     def frompath(path):
         with open(path, 'rb') as f:
             magic = f.read(4)
+            if type(magic) is bytes:
```
Comment (jparise) [inline BazelExtensions/headermap_tool.py:34]:
```suggestion
            if isinstance(magic, bytes):
```

File: BazelExtensions/headermap_tool.py [hunk 2]
```diff
@@ -84,6 +86,8 @@ def frompath(path):
                 raise SystemExit("error: %s: unable to read zero-sized string table"%(
                         path,))
             strtable = f.read(strtable_size)
+            if type(strtable) is bytes:
```
Comment (jparise) [inline BazelExtensions/headermap_tool.py:89]:
```suggestion
            if isinstance(strtable, bytes):
```

File: bin/update_pods.py [hunk 3]
```diff
@@ -203,7 +203,7 @@ def _needs_update(invocation_info):
     pod_root_dir = repository_ctx.GetPodRootDir()
     _exec(repository_ctx, ["/bin/bash", "-c", "mkdir -p " + pod_root_dir])
     _exec(repository_ctx, ["/bin/bash", "-c", "touch .pod-version"], pod_root_dir)
-    cached_version = _exec(repository_ctx, ["/bin/bash", "-c", "cat .pod-version"], pod_root_dir).split("\n")[0]
+    cached_version = _exec(repository_ctx, ["/bin/bash", "-c", "cat .pod-version"], pod_root_dir).split(b'\n')[0]
```
Comment (jparise) [inline bin/update_pods.py:206]:
```suggestion
    cached_version = _exec(repository_ctx, ["/bin/bash", "-c", "cat .pod-version"], pod_root_dir).splitlines()[0]
```

**Review by OliverKoo — APPROVED** [review]

File: BazelExtensions/acknowledgement_merger.py [hunk 4]
```diff
@@ -1,8 +1,7 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
```
Comment (woshimaliang) [inline BazelExtensions/acknowledgement_merger.py:1]: #!/usr/bin/**env python3** ?

Comment (garrettmoon) [reply BazelExtensions/acknowledgement_merger.py:1]: I don't know anything, but this is why I landed on /usr/bin/python3: https://blogs.gnome.org/mcatanzaro/2018/02/16/on-python-shebangs/

Comment (woshimaliang) [reply BazelExtensions/acknowledgement_merger.py:1]: me neither. i just read https://github.com/pinterest/PodToBUILD/pull/200#pullrequestreview-929198622

Comment (luispadron) [reply BazelExtensions/acknowledgement_merger.py:1]: So I believe we should use `env` here, I dont see where in that article it says not to use `env`. I think they're mostly talking about not specifying a version of python (which this PR does specify).

On my machine:

```
/usr/bin/python3 --version
> 3.8.9

/usr/bin/env python3 --version
> 3.10.2
``

The `env` here will prefer to use any environment versions of python over the system installed one which is what most people would expect.

Comment (garrettmoon) [reply BazelExtensions/acknowledgement_merger.py:1]: You've convinced me 😅 I'll update them.

**Review by woshimaliang — APPROVED** [review]

**Pull request merged**